src/levanter/data/text.py

```diff
@@ -502,8 +562,14 @@ class LMDatasetConfig:
     def the_tokenizer(self) -> PreTrainedTokenizerFast:
         return load_tokenizer(self.tokenizer)
 
-    def build_or_load_cache(self, split: str, monitors: Union[bool, List[MetricsMonitor]] = True):
-        source = self.get_shard_source(split)
+    def build_or_load_cache(
+        self, split: str, monitors: Union[bool, List[MetricsMonitor]] = True
+    ) -> Optional[TokenizedDocumentCache]:
```

Comment:
This is for handling cases when the dev split is not available (RP dataset, for example)